pkg/appregistry/adapter.go

```diff
@@ -0,0 +1,71 @@
+package appregistry
+
+import (
+	"bytes"
+
+	appr "github.com/operator-framework/go-appr/appregistry"
+	appr_blobs "github.com/operator-framework/go-appr/appregistry/blobs"
+	appr_package "github.com/operator-framework/go-appr/appregistry/package_appr"
+	appr_models "github.com/operator-framework/go-appr/models"
+)
+
+const (
+	mediaType string = "helm"
+)
+
+// This interface (internal to this package) encapsulates nitty gritty details of go-appr client bindings
+type apprApiAdapter interface {
+	// ListPackages returns a list of package(s) available to the user
+	ListPackages() (appr_models.Packages, error)
+
+	// GetPackageMetadata returns metadata associated with a given package
+	GetPackageMetadata(namespace string, repository string, release string) (*appr_models.Package, error)
+
+	// DownloadOperatorManifest downloads the blob associated with a given digest that directly corresponds to a package release
+	DownloadOperatorManifest(namespace string, repository string, digest string) ([]byte, error)
+}
+
+type apprApiAdapterImpl struct {
+	client *appr.Appregistry
+}
+
+func (a *apprApiAdapterImpl) ListPackages() (appr_models.Packages, error) {
+	params := appr_package.NewListPackagesParams()
```

Comment:
Don't we allow listing packages within a namespace? I think we should allow some filtering here

Reply:
I already have a new PR in review that allows to filter by namespace.